autorest/azure/rp.go

```diff
@@ -60,7 +57,7 @@ func DoRetryWithRegistration(client autorest.Client) autorest.SendDecorator {
 				}
 				err = re
 
-				if re.ServiceError != nil && re.ServiceError.Code == "MissingSubscriptionRegistration" {
+				if re.ServiceError != nil && re.ServiceError.Code == "MissingSubscriptionRegistration" && !client.SkipResourceProviderRegistration {
```

Comment:
> && !client.SkipResourceProviderRegistration  [](start = 90, length = 45)

is there something we cna add to the `rp_test.go` for this scenario ?

Reply:
Good call, and adding a test made me realize the fix is wrong.  :/